code/controllers/CMSMain.php

```diff
@@ -1316,7 +1316,6 @@ public function duplicate($request) {
 		
 		if(($id = $this->urlParams['ID']) && is_numeric($id)) {
 			$page = DataObject::get_by_id("SiteTree", $id);
-			if($page && (!$page->canEdit() || !$page->canCreate())) return Security::permissionFailure($this);
```

Comment:
canCreate shoud have all the necessary logic to perform this check.

```php
$member = Member::currentUser();
$page = $this;
$parent = $page->Parent();
$context = array('Parent' => $parent);
$canDuplicate = singleton($page->ClassName)->canCreate($member, $context);
```

You could even add `'Duplicate' => $page` to the context if you wanted, although this would be 3.2 land.